common/src/main/java/io/netty/util/HashedWheelTimer.java

```diff
@@ -706,6 +750,7 @@ public void remove(HashedWheelTimeout timeout) {
             timeout.prev = null;
             timeout.next = null;
             timeout.bucket = null;
+            timeout.timer.pendingTimeouts.decrementAndGet();
```

Comment:
Actually, I would still prefer to have pendingTimeout counter maintained so that clients of HashWheel could use the public pendingTimeoutsCount method for monitoring/their own guards.